jerry-core/ecma/builtin-objects/ecma-builtin-array.inc.h

```diff
@@ -19,6 +19,8 @@
 
 #include "ecma-builtin-helpers-macro-defines.inc.h"
 
+#ifndef CONFIG_DISABLE_ARRAY_BUILTIN
+
```

Comment:
These `#ifndef` guards are what the commit message is referring to by saying:

> The commit also changes some builtin-related .inc.h files by
> adding guards that don't affect functionality but improve the
> results of the generator.

So, yes indeed, they are "pointless". But for the script that generates the `lit-magic-strings.inc.h` file, they are of big help, because that script analyses each `.c` and `.h` file in the project separately, which means that it cannot infer the guard context that is on the including side of a header. Thus, if these guards are not added then the generator script sees all magic string references in these `.inc.h` files as unguarded and decides that their definition must always be present not to break a build.

Right now, I see no other approach that could work, other than writing a context-sensitive (include-tracking) analyser -- but writing something like that just for this task would be an overkill IMHO.

Repeating these guards in these `.inc.h` files is a bit of duplication, I agree, but they are not completely illogical either, and I'd say they don't seem to be too hard to keep in sync with the guards of the including source files.